Themis/ViewModels/Assessment/CodeEditorViewModel.swift

```diff
@@ -49,6 +49,21 @@ class CodeEditorViewModel: ObservableObject {
         return nil
     }
     
+    private var allFiles: [Node] {
+        var files: [Node] = []
+        var nodesToCheck = fileTree
+        while !nodesToCheck.isEmpty {
+            let currentNode = nodesToCheck.removeFirst()
+            if currentNode.type == .file {
+                files.append(currentNode)
+            }
+            if let children = currentNode.children {
+                nodesToCheck.append(contentsOf: children)
+            }
+        }
+        return files
+    }
```

Comment:
Put this into Node (probably rename it to make it more clear like recursiveChildren or so), im sure we will need it somewhere else as well.